core/src/main/java/jenkins/security/BasicHeaderProcessor.java

```diff
@@ -125,11 +125,7 @@ protected boolean authenticationIsRequired(String username) {
         // BASIC protocol is desirable. This behaviour is also consistent with that provided by form and digest,
         // both of which force re-authentication if the respective header is detected (and in doing so replace
         // any existing AnonymousAuthenticationToken). See SEC-610.
-        if (existingAuth instanceof AnonymousAuthenticationToken) {
-            return true;
-        }
-
-        return false;
+        return existingAuth instanceof AnonymousAuthenticationToken;
```

Comment:
This change makes the preceding comment much more difficult to understand.

Reply:
Hi, I suggest you can use (Ctrl+I). This shortcut evaluates any code statement. We must care about code readability.

Reply:
I'm confused by this comment.

Reply:
This statement says if existingAuth object is derived AnonymousAuthenticationToken return true, else return false. I removed only if statement.

Reply:
I know what you did. I'm confused by your comment 

> Hi, I suggest you can use (Ctrl+I). This shortcut evaluates any code statement. We must care about code readability.

in response to me pointing out that the preceding source code comment makes no sense after this change.